### ISSUE TITLE
Angry Pyromaniac Survivor can't be Stopped

### ISSUE THREAD

**Issue by coyo7e** [CleverRaven/Cataclysm-DDA#54263]
### Describe the bug

I recently ran into a (I think) new type of survivor, the "angry pyromaniac," who wants the player to go burn down a specific building. I was like, "cool fine, no problem!" and then realized he'd begun following me, and when I went to speak to him I couldn't do anything but say "OK" to him with two variants and then close the dialogue. 

This became a problem quickly, because I hopped into a vehicle and he kept vocalizing about nearby corpses which he then tried to jump out of the moving vehicle to smash (as long as you're moving they stay on the running boards/doorway and won't jump out, at least). Additionally since he seemed to be using the "stay within 4 spaces" behavior by default he'd rubber band back to me constantly and keep vocalizing, and eventually he started a fight with about 15 zeds and a mi-go at once after hearing a noise, because I was simply unable to stop him from doing anything or tell him to guard a spot, etc.

This survivor quest seems fun (although I have reservations about the time required to burn down a building, since it can take days sometimes) however, it's kind of unreasonable for someone to try and drag this chaotically-behaving NPC to the target and not have them get killed on the way. Alternatively, the conversation dialogue might warn the player that he'll immediately follow them so they could at least turn him down and come back later after clearing out a path (but you can't find the path until you pick up the quest, so chicken and egg problem..)

I am not sure why it was decided to have the NPC follow the player outside of some flavor stuff of them wanting proof of the burned building, since, what's the point of having a social-based character if you can't lie to an NPC about finishing their quests, etc? I even tried to temporarily ditch him while he was pulping a corpse, and he beelined after me and caught up.

### Steps to reproduce

1. Speak to Angry Pyromaniac NPC. 
2. Accept their quest.
3.  Try to talk to them about anything/go near enemies.

### Expected behavior

I had expected to somehow be able to rein in this NPC during the quest, since I had no warning and had to drag him into a city I'd never explored before as soon as I picked up the quest.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 939ec93 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Graphical Overmap [Graphical_Overmap],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

**Comment by NetSysFire** [comment]
I got that in an earlier run. Very, very bad when they block the only exit of a room or building. And if you try to make them stay there and go out through the window, they will smash said window.

**Comment by Ramza13** [comment]
I think this might just be #54250

**Comment by NetSysFire** [comment]
No, this is a different NPC. The other pyromaniac wants you to actively burn down stuff and follows you, the arsonist with their fluffy ears does not. I am currently trying to track down the exact NPC.

**Comment by NetSysFire** [comment]
It appears to be the pizzaiolo NPC:

https://github.com/CleverRaven/Cataclysm-DDA/blob/5458faf562e7769aafe387ccebfd043970d33fe6/data/json/npcs/other/NPC_pizzaiolo.json

**Comment by coyo7e** [comment]
> It appears to be the pizzaiolo NPC:
> 
> https://github.com/CleverRaven/Cataclysm-DDA/blob/5458faf562e7769aafe387ccebfd043970d33fe6/data/json/npcs/other/NPC_pizzaiolo.json

Yup!

**Comment by LeahLuong** [comment]
Is he responsive to temp orders using the "C" menu??

**Comment by I-am-Erk** [comment]
The option to convince him to stay somewhere safe while you "burn the building" and then come back and lie to him would be a nice one. Most of our mission lines want the option to lie when it doens't make sense, this is one where it actually might. That would require a fair bit more dialogue and scripting though.

**Comment by coyo7e** [comment]
> The option to convince him to stay somewhere safe while you "burn the building" and then come back and lie to him would be a nice one. Most of our mission lines want the option to lie when it doens't make sense, this is one where it actually might. That would require a fair bit more dialogue and scripting though.
Technically it would probably require less overall than the NPC currently wants. He insists on following right now, I've seen a couple posts on reddit where people got stuck in a bathroom/etc because he wouldn't leave the doorway.

> Is he responsive to temp orders using the "C" menu??
No. He's unresponsive to orders, period. That was most of my point.
